third_party/silabs/slc/exporter_templates/platform_library/custom.Makefile

```diff
@@ -114,25 +107,80 @@ target_sources({{PROJECT_NAME}} PRIVATE
     {%- set source = prepare_path(source) -%}
 
     {#- Only take PAL sources #}
-    {%- if ('${PROJECT_SOURCE_DIR}/src/src' in source) -%}
         {%- if source.endswith('.s') or source.endswith('.S') %}
 target_sources({{PROJECT_NAME}} PRIVATE {{source}})
 set_property(SOURCE {{source}} PROPERTY LANGUAGE C)
         {%- endif %}
-    {%- endif %}
 {%- endfor %}
 
 # ==============================================================================
 # Compile definitions
 # ==============================================================================
+
+target_compile_definitions({{PROJECT_NAME}}-config INTERFACE
+{%- for define in C_CXX_DEFINES %}
+    {%- if not (define.startswith("MBEDTLS_")
+                or define.startswith("OPENTHREAD_")
+                ) %}
+    {{define}}={{C_CXX_DEFINES[define]}}
+    {%- endif %}
+{%- endfor %}
+)
+
+{% if dict_contains_key_starting_with(C_CXX_DEFINES, "OPENTHREAD_") -%}
 target_compile_definitions(ot-config INTERFACE
 {%- for define in C_CXX_DEFINES %}
-    {%- if not ( define.startswith("MBEDTLS_PSA_CRYPTO_CLIENT") or ("OPENTHREAD_RADIO" == define) or ("OPENTHREAD_FTD" == define) or ("OPENTHREAD_MTD" == define) or ("OPENTHREAD_COPROCESSOR" == define) ) %}
+    {%- if define.startswith("OPENTHREAD_")
+            and not ( ("OPENTHREAD_RADIO" == define)
+                or ("OPENTHREAD_FTD" == define)
+                or ("OPENTHREAD_MTD" == define)
+                or ("OPENTHREAD_COPROCESSOR" == define) ) %}
+    {{define}}={{C_CXX_DEFINES[define]}}
+    {%- endif %}
+{%- endfor %}
+)
+
+{% endif -%}
+
+{% if dict_contains_key_starting_with(C_CXX_DEFINES, "MBEDTLS_")
+        or dict_contains_key_starting_with(C_CXX_DEFINES, "PSA_")
+        -%}
+target_compile_definitions({{PROJECT_NAME}}-mbedtls-config INTERFACE
+{%- for define in C_CXX_DEFINES %}
+    {%- if define.startswith("MBEDTLS_") or
+           define.startswith("PSA_")
+     %}
     {{define}}={{C_CXX_DEFINES[define]}}
     {%- endif %}
 {%- endfor %}
 )
 
+{% endif -%}
+
+target_include_directories({{PROJECT_NAME}}-mbedtls-config INTERFACE
+    autogen
+    config
+)
+
+target_link_libraries({{PROJECT_NAME}}-mbedtls PRIVATE {{PROJECT_NAME}}-mbedtls-config)
+target_link_libraries({{PROJECT_NAME}} PRIVATE {{PROJECT_NAME}}-mbedtls-config)
+
+{% if PROJECT_NAME.startswith("openthread-efr32-rcp") -%}
+target_link_libraries(ot-config-radio INTERFACE {{PROJECT_NAME}}-mbedtls-config)
+target_link_libraries(ot-config-radio INTERFACE {{PROJECT_NAME}}-config)
+{% elif PROJECT_NAME.startswith("openthread-efr32-mtd") -%}
+target_link_libraries(ot-config-ftd INTERFACE {{PROJECT_NAME}}-mbedtls-config)
+target_link_libraries(ot-config-ftd INTERFACE {{PROJECT_NAME}}-config)
```

Comment:
Are we linking against correct libraries `ot-config-**ftd**` for `openthread-efr32-mtd` or it should be `ot-config-mtd`?